app/assets/javascripts/ordered_creator.js

```diff
@@ -1,12 +1,45 @@
+function bindUpDownArrow() {
+  $('.ordered-up-arrow').unbind("click").on('click', function(e) {
+    swapUp(this);
+  });
+  $('.ordered-down-arrow').unbind("click").on('click', function(e) {
+    swapDown(this);
+  });
+}
+
+function swapUp(selector) {
+  $(selector).parent().insertBefore($(selector).parent().prev());
+  reindexNestedOrderedField();
+}
+
+function swapDown(selector) {
+  $(selector).parent().insertAfter($(selector).parent().next());
+  reindexNestedOrderedField();
+}
+
 function orderedDragAndDrop(selector) {
-  selector.nestable({maxDepth: 0, rootClass: 'ordered-field-container'});
-  selector.on('change', function(event) {
-    hidden_index_selectors = $(this).find(".index");
-    hidden_index_selectors.each(function( index ) {
-      hidden_index_selectors[index].value = index;
-    });
+  selector.nestable({ maxDepth: 0, rootClass: 'ordered-field-container' });
+  selector.on('change', function (event) {
+    reindexNestedOrderedField();
+  });
+}
+
+function reindexNestedOrderedField(mutationsList) {
+  hidden_index_selectors = $($(".ordered-field-container .index"));
+  hidden_index_selectors.each(function (index) {
+    hidden_index_selectors[index].value = index;
   });
+  bindUpDownArrow();
```

Comment:
I think this should be moved out of this method, then a new mutationobserver created to handle running this method, and then the method ran one time inside the Blacklight.onload (to bind all of the pre-existing rows)

Reply:
I was thinking of doing that and had it in there at one point. I wasnt sure if we wanted to keep creating mutation observers or just piggyback off the one.